modules/conversantAnalyticsAdapter.js

```diff
@@ -0,0 +1,470 @@
+import {ajax} from '../src/ajax.js';
+import adapter from '../src/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+import {getGlobal} from '../src/prebidGlobal.js';
+import adapterManager from '../src/adapterManager.js';
+import * as utils from '../src/utils.js';
+
+const {
+  EVENTS: { AUCTION_END, AD_RENDER_FAILED, BID_TIMEOUT, BID_WON }
+} = CONSTANTS;
+
+const ANALYTICS_TYPE = 'endpoint';
+// for local testing set domain to 127.0.0.1:8290
+const URL = 'https://web.hb.ad.cpe.dotomi.com/cvx/event/prebidanalytics';
+const ANALYTICS_CODE = 'conversant';
+export const LOG_PREFIX = 'Conversant analytics adapter: ';
+
+// Maximum time to keep an item in the cache before it gets purged
+export const MAX_MILLISECONDS_IN_CACHE = 30000;
+
+// How often cache cleanup will run
+export const CACHE_CLEANUP_TIME_IN_MILLIS = 30000;
+
+// BID STATUS CODES
+export const CNVR_WIN = 10;
+export const CNVR_BID = 20;
+export const CNVR_NO_BID = 30;
+export const CNVR_TIMEOUT = 40;
+export const CNVR_RENDER_FAILED = 50;
```

Comment:
Let's reduce the number of exports, and organize them into 2: one for constants and one fo states.  This could also help with the double import of this file in the test.

```javascript
export const CNVR_CONSTANTS = {
   STATUS: {
      WIN: 10,
      BID: 20, 
      ...
   }, 
   MAX_MILLISECONDS_IN_CACHE: 30000,
   CACHE_CLEANUP_TIME_IN_MILLIS: 30000,
   ...
};

export const cnvr_state = {
   doSample: true,
   adIdLookup: {},
   timeoutCache: {},
   ...
};
```

Reply:
I tried to implement this change and after getting 1/2 through it really didn't feel right.  For testing we need to "import *" to get access to the functions so we will always have a double import.  Secondly, by nesting all those constants in an object we actually remove the fact that we want them to be constant and allow them to be re-assigned.  Finally, some of those constants, by some weird Javascript stuff does not allow it to be used when it's nested in an object like that so we end up with even more piecemeal where some are in the object, some are not and I'll still want to have all the documentation associated with them so it won't just be one small compact object.

Reply:
The functions should be added into a single object for export as well.  The constant thing unfortunately is just how js works.  Hence it's important not to change all-caps variables.  Take a look at https://github.com/conversant/Prebid.js/pull/4.  It organizes the exports and still retain comments.